### PR TITLE
06 - Reject game handler

### PR DESCRIPTION
**Pull request by xavierlepretre**

### CODE REVIEW

File: x/checkers/keeper/msg_server_reject_game.go [hunk 1]
```diff
@@ -10,8 +12,35 @@ import (
 func (k msgServer) RejectGame(goCtx context.Context, msg *types.MsgRejectGame) (*types.MsgRejectGameResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Handling the message
-	_ = ctx
+	storedGame, found := k.Keeper.GetStoredGame(ctx, msg.IdValue)
+	if !found {
+		return nil, errors.New("Game not found " + msg.IdValue)
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_reject_game.go:17]: sdkerrors if possible

File: x/checkers/keeper/msg_server_reject_game.go [hunk 2]
```diff
@@ -10,8 +12,35 @@ import (
...
+	}
+	fullGame := storedGame.ToFullGame()
+
+	// Is it an expected player? And did the player already play?
+	if strings.Compare(storedGame.Red, msg.Creator) == 0 {
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_reject_game.go:22]:
```suggestion
	if strings.Compare(storedGame.Red, msg.Creator) {
```

File: x/checkers/keeper/msg_server_reject_game.go [hunk 3]
```diff
@@ -10,8 +12,35 @@ import (
...
+		if 1 < fullGame.MoveCount {
+			return nil, errors.New("Red player has already played, and cannot reject")
+		}
+	} else if strings.Compare(storedGame.Black, msg.Creator) == 0 {
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_reject_game.go:26]:
```suggestion
	} else if strings.Compare(storedGame.Black, msg.Creator) {
```

File: x/checkers/keeper/msg_server_reject_game.go [hunk 4]
```diff
@@ -10,8 +12,35 @@ import (
...
+	storedGame, found := k.Keeper.GetStoredGame(ctx, msg.IdValue)
+	if !found {
+		return nil, errors.New("Game not found " + msg.IdValue)
+	}
+	fullGame := storedGame.ToFullGame()
+
+	// Is it an expected player? And did the player already play?
+	if strings.Compare(storedGame.Red, msg.Creator) == 0 {
+		if 1 < fullGame.MoveCount {
+			return nil, errors.New("Red player has already played, and cannot reject")
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_reject_game.go:24]: sdkerrors if possible

File: x/checkers/keeper/msg_server_reject_game.go [hunk 5]
```diff
@@ -10,8 +12,35 @@ import (
...
+		}
+	} else if strings.Compare(storedGame.Black, msg.Creator) == 0 {
+		if 0 < fullGame.MoveCount {
+			return nil, errors.New("Black player has already played, and cannot reject")
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_reject_game.go:28]: ditto

File: x/checkers/keeper/msg_server_reject_game.go [hunk 6]
```diff
@@ -10,8 +12,35 @@ import (
...
+	} else {
+		return nil, errors.New("Message creator is not a player")
+	}
+
+	// Remove the game completely as it is not interesting to keep it.
+	k.Keeper.RemoveStoredGame(ctx, msg.IdValue)
+
+	// What to emit
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(types.RejectGameEventKey,
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_reject_game.go:39]: see https://github.com/cosmos/b9-checkers-academy-draft/pull/5 for event construction

File: x/checkers/types/full_game.go [hunk 7]
```diff
@@ -20,6 +23,7 @@ func (fullGame *FullGame) ToStoredGame() (storedGame *StoredGame) {
 	storedGame.Turn = fullGame.Game.Turn.Color
 	storedGame.Red = fullGame.Red.String()
 	storedGame.Black = fullGame.Black.String()
+	storedGame.MoveCount = strconv.FormatUint(fullGame.MoveCount, 10)
```
Comment (marbar3778) [inline x/checkers/types/full_game.go:26]: i believe this is an unnecessary conversion. the value is already a uint64

File: x/checkers/types/full_game.go [hunk 8]
```diff
@@ -35,6 +39,7 @@ func (storedGame *StoredGame) ToFullGame() (fullGame *FullGame) {
 	fullGame.Game = *game
 	fullGame.Red, err = sdk.AccAddressFromBech32(storedGame.Red)
 	fullGame.Black, err = sdk.AccAddressFromBech32(storedGame.Black)
+	fullGame.MoveCount, err = strconv.ParseUint(storedGame.MoveCount, 10, 64)
```
Comment (marbar3778) [inline x/checkers/types/full_game.go:42]: ditto

File: proto/checkers/stored_game.proto [hunk 9]
```diff
@@ -12,4 +12,5 @@ message StoredGame {
   string turn = 4;
   string red = 5; 
   string black = 6; 
+  string moveCount = 7;
```
Comment (marbar3778) [inline proto/checkers/stored_game.proto:15]: why string and not uint64

**Review by marbar3778 — APPROVED** [review]

File: proto/checkers/stored_game.proto (hunk 9 above)
Comment (xavierlepretre) [reply proto/checkers/stored_game.proto:15]: Yes, no reason. Let me change it.

**Pull request closed without merging**